File: example/marker_demo/marker_demo.f90
program marker_demo
    !! Demonstrates different marker types and styles
    
    use fortplot
    use, intrinsic :: iso_fortran_env, only: wp => real64
    implicit none
    
    call demo_scatter_plot()
    call demo_all_marker_types()
    call demo_marker_colors()
    
    print *, "=== Marker Examples ==="
    print *, "Created: scatter_plot.png/pdf/txt"
    print *, "Created: all_marker_types.png/pdf/txt"
    print *, "Created: marker_colors.png/pdf/txt"
    print *, "All marker examples completed!"

contains


    subroutine demo_scatter_plot()
        !! Creates a scatter plot to demonstrate markers in practical use
        type(figure_t) :: fig
        real(wp) :: x(20), y(20)
        integer :: i
        
        ! Generate scatter data
        do i = 1, 20
            x(i) = real(i, wp) * 0.3_wp
            y(i) = sin(x(i)) + 0.1_wp * real(i - 10, wp) / 10.0_wp  ! Sine wave with trend
        end do
        
        call fig%initialize(600, 450)
        call fig%set_title("Scatter Plot with Antialiased Markers")
        call fig%set_xlabel("X Values")
        call fig%set_ylabel("Y Values")
        
        ! Scatter plot with markers
        call fig%add_plot(x, y, marker='o', linestyle='None', label='Data Points')
        
        ! Add trend line for context
        call fig%add_plot(x, sin(x), linestyle='-', label='Sin(x) Reference')
        
        call fig%legend()
        call fig%savefig('example/marker_demo/scatter_plot.png')
        call fig%savefig('example/marker_demo/scatter_plot.pdf')
        call fig%savefig('example/marker_demo/scatter_plot.txt')
    end subroutine demo_scatter_plot

    subroutine demo_all_marker_types()
        !! Demonstrates all available marker types with realistic data
        type(figure_t) :: fig
        real(wp) :: x1(10), y1(10), x2(10), y2(10), x3(10), y3(10), x4(10), y4(10)
        integer :: i
        
        ! Generate realistic data for each marker type
        do i = 1, 10
            x1(i) = real(i, wp) * 0.5_wp
            y1(i) = sin(x1(i)) + 3.0_wp
            x2(i) = real(i, wp) * 0.5_wp
            y2(i) = cos(x2(i)) + 2.0_wp
            x3(i) = real(i, wp) * 0.5_wp
            y3(i) = sin(x3(i) * 2.0_wp) + 1.0_wp
            x4(i) = real(i, wp) * 0.5_wp
            y4(i) = cos(x4(i) * 1.5_wp)
        end do
        
        call fig%initialize(600, 400)
        call fig%set_title("All Marker Types")
        call fig%set_xlabel("X Values")
        call fig%set_ylabel("Y Values")
        
        ! Draw each marker type with data
        call fig%add_plot(x1, y1, marker='o', linestyle='None', label='Circle')
        call fig%add_plot(x2, y2, marker='s', linestyle='None', label='Square')
        call fig%add_plot(x3, y3, marker='D', linestyle='None', label='Diamond')
        call fig%add_plot(x4, y4, marker='x', linestyle='None', label='Cross')
        
        call fig%legend()
        call fig%savefig('example/marker_demo/all_marker_types.png')
        call fig%savefig('example/marker_demo/all_marker_types.pdf')
        call fig%savefig('example/marker_demo/all_marker_types.txt')
    end subroutine demo_all_marker_types

    subroutine demo_marker_colors()
        !! Demonstrates different colored markers with realistic data
        type(figure_t) :: fig
        real(wp) :: x1(8), y1(8), x2(8), y2(8), x3(8), y3(8)
        integer :: i
        
        ! Generate realistic data for color demonstration
        do i = 1, 8
            x1(i) = real(i, wp) * 0.6_wp
            y1(i) = exp(-x1(i) * 0.3_wp) * cos(x1(i)) + 2.5_wp
            x2(i) = real(i, wp) * 0.6_wp
            y2(i) = exp(-x2(i) * 0.2_wp) * sin(x2(i)) + 1.5_wp
            x3(i) = real(i, wp) * 0.6_wp
            y3(i) = exp(-x3(i) * 0.4_wp) * sin(x3(i) * 1.5_wp) + 0.5_wp
        end do
        
        call fig%initialize(500, 400)
        call fig%set_title("Marker Colors and Styles")
        call fig%set_xlabel("X Position")
        call fig%set_ylabel("Y Position")
        
        ! Different marker types with automatic color cycling
        call fig%add_plot(x1, y1, marker='o', linestyle='None', label='Blue circles')
        call fig%add_plot(x2, y2, marker='s', linestyle='None', label='Green squares')
        call fig%add_plot(x3, y3, marker='D', linestyle='None', label='Orange diamonds')
        
        call fig%legend()
        call fig%savefig('example/marker_demo/marker_colors.png')
        call fig%savefig('example/marker_demo/marker_colors.pdf')
        call fig%savefig('example/marker_demo/marker_colors.txt')
    end subroutine demo_marker_colors

end program marker_demo

File: example/python/marker_demo/marker_demo.py
#!/usr/bin/env python3
"""Demonstrates different marker types and styles - Dual mode: fortplot or matplotlib"""

import sys
import numpy as np

# Dual-mode import: --matplotlib uses matplotlib, default uses fortplot
if "--matplotlib" in sys.argv:
    import matplotlib.pyplot as plt
    backend = "matplotlib"
else:
    import fortplot.fortplot as plt
    backend = "fortplot"

def demo_scatter_plot():
    """Creates a scatter plot to demonstrate markers in practical use"""
    print(f"=== Scatter Plot Demo ({backend}) ===")
    
    # Generate scatter data
    x = np.arange(1, 21) * 0.3
    y = np.sin(x) + 0.1 * (np.arange(1, 21) - 10) / 10.0  # Sine wave with trend
    
    plt.figure(figsize=(7.5, 5.625))
    plt.title("Scatter Plot with Antialiased Markers")
    plt.xlabel("X Values")
    plt.ylabel("Y Values")
    
    # Simplified plots for dual-mode compatibility
    plt.plot(x, y, linestyle="", label='Data Points')  # No line, just markers
    plt.plot(x, np.sin(x), linestyle="-", label='Sin(x) Reference')
    
    plt.legend()
    plt.savefig('scatter_plot.png')
    plt.savefig('scatter_plot.pdf')
    
    # Save TXT for fortplot only
    if backend == "fortplot":
        plt.savefig('scatter_plot.txt')
    
    if backend == "matplotlib":
        plt.close()
    
    print("Created: scatter_plot.png/pdf" + ("" if backend == "matplotlib" else "/txt"))

def demo_all_marker_types():
    """Demonstrates all available marker types with realistic data"""
    # Generate realistic data for each marker type
    x = np.arange(1, 11) * 0.5
    y1 = np.sin(x) + 3.0
    y2 = np.cos(x) + 2.0
    y3 = np.sin(x * 2.0) + 1.0
    y4 = np.cos(x * 1.5)
    
    fig, ax = plt.subplots(figsize=(7.5, 5))
    ax.set_title("All Marker Types")
    ax.set_xlabel("X Values")
    ax.set_ylabel("Y Values")
    
    # Draw each marker type with data (pyplot-fortran style)
    ax.plot(x, y1, 'o', label='Circle')
    ax.plot(x, y2, 's', label='Square')
    ax.plot(x, y3, 'D', label='Diamond')
    ax.plot(x, y4, 'x', label='Cross')
    
    ax.legend()
    plt.savefig('all_marker_types.png', dpi=100)
    plt.savefig('all_marker_types.pdf')
    plt.close()

def demo_marker_colors():
    """Demonstrates different colored markers with realistic data"""
    # Generate realistic data for color demonstration
    x = np.arange(1, 9) * 0.6
    y1 = np.exp(-x * 0.3) * np.cos(x) + 2.5
    y2 = np.exp(-x * 0.2) * np.sin(x) + 1.5  
    y3 = np.exp(-x * 0.4) * np.sin(x * 1.5) + 0.5
    
    fig, ax = plt.subplots(figsize=(6.25, 5))
    ax.set_title("Marker Colors and Styles")
    ax.set_xlabel("X Position")
    ax.set_ylabel("Y Position")
    
    # Different marker types with automatic color cycling (pyplot-fortran style)
    ax.plot(x, y1, 'o', label='Blue circles')
    ax.plot(x, y2, 's', label='Green squares')
    ax.plot(x, y3, 'D', label='Orange diamonds')
    
    ax.legend()
    plt.savefig('marker_colors.png', dpi=100)
    plt.savefig('marker_colors.pdf')
    plt.close()

if __name__ == "__main__":
    demo_scatter_plot()
    # Other demos need conversion from subplots to simple pyplot
    print("Marker demo completed!")